### DISCUSSION TITLE
How do I include "References" and "Citation" section in the TOC?

### DISCUSSION THREAD

**Discussion by tbenthompson** [quarto-dev/quarto-cli#6021] [Q&A] [answered]
### Description

For HTML output formats, the automatically generated "References" and "Citation" sections are not included in the table of contents. Is there a way to include those sections in the TOC? 

Thanks!

**Comment by cderv** [comment, marked as answer]
This is because References and citations have a special treatment in our Default template to be included into the appendix section at the bottom. See https://quarto.org/docs/authoring/appendices.html


You could opt out this for it to behave like regular document with References place by default at the bottom of the document. You could then add a section above and it would be in the TOC 

````markdown
---
title: "My Document"
format: html
toc: true
appendix-style: none
bibliography: references.bib
---

# Title

```{r}
knitr::write_bib("knitr", "references.bib")
```

# References 

See @R-knitr
````

You can also control the placement of the bibliography. This will prevent the References part to be included in the appendix
````markdown
---
title: "My Document"
format: html
toc: true
bibliography: references.bib
---

# References

::: {#refs}
:::

# Title

```{r}
#| include: false
knitr::write_bib("knitr", "references.bib")
```

See @R-knitr


````

Hope it helps

**Comment by tbenthompson** [comment]
Thanks so much! 

Two more questions if you don't mind:
- Is there also a way to include the "Citation" section in the TOC? This is what I was referring to in the original message:
<img width="765" alt="2023-06-26 10 34 33@2x" src="https://github.com/quarto-dev/quarto-cli/assets/4241811/d9641472-ae7f-4049-a03e-a3683b2f29fc">


- Ideally, I'd have these appendices (References and Citation both) display with smaller text in the TOC. Is there a template or something that I could modify to achieve that? I looked into using [partials](https://quarto.org/docs/journals/templates.html#template-partials) and found the [TOC partial](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/html/pandoc/toc.html). But it looks like the actual contents of the TOC are buried a bit deeper.

Thanks again.

**Reply by cderv** [reply]
> Is there also a way to include the "Citation" section in the TOC?

I don't think there is. @dragonstyle should we maybe see if we can add those headers to TOC when `appendix-style: plain` is set ? 

> Ideally, I'd have these appendices (References and Citation both) display with smaller text in the TOC. Is there a template or something that I could modify to achieve that? 

Or @dragonstyle do you think this is an improvement we can do related to inclusion and style maybe ? Just not sure if we are building the appendix part too late maybe... 

If this is something we could try to have a look at, I can create an issue to track potential improvment ?

**Reply by tbenthompson** [reply]
Thanks! I would treat these things as nice-to-haves rather than requirements. Quarto is working wonderfully for us.
